src/awesome/awesome.rs

```diff
@@ -94,10 +94,33 @@ fn register_xproperty<'lua>(_: &'lua Lua, _: Value<'lua>) -> rlua::Result<()> {
 
 /// Get layout short names
 fn xkb_get_group_names<'lua>(_: &'lua Lua, _: ()) -> rlua::Result<String> {
+    use std::process::{Stdio, Command};
+    // TODO FIXME wow this is _really_ _really_ dumb.
+    // like really dumb
+    // but not sure how else to do this to communicate to xwayland...
     warn!("xkb_get_group_names not supported");
-    warn!("Returning dummy data so I can move forward");
-    warn!("Replace me before release!");
-    Ok("pc+us+inet(evdev)".into())
+    warn!("Doing a bad hack now");
+    warn!("Fix me before release!");
+    // This should be a function, not a whole freaking command!
+    Ok(Command::new("/usr/bin/setxkbmap").arg("-v").arg("-query")
+        .stdout(Stdio::piped())
+        .spawn()
+        .map(|child| {
+            child.wait_with_output().ok().and_then(|output| {
+                let stdout = String::from_utf8(output.stdout)
+                    .expect("Output was not utf8");
+                for line in stdout.lines() {
+                    if line.contains("symbols:") {
+                        let last = line.split(" ").last().unwrap();
+                        return Some(last.trim().into())
+                    }
+                }
+                None
+            })
+        }).unwrap_or_else(|err| {
+            warn!("err: {:#?}", err);
+            Some("pc+us+inet(evdev)".into())
+        }).unwrap())
```

Comment:
I liked your previous "just return static fake data" version of this more. :-)
On a more serious note, I would expect a wayland compositor to already have the required data here somewhere. From a quick look at wlroots, this data seems to be [in struct roots_keyboard_config](https://github.com/swaywm/wlroots/blob/a081b3bf7b99c0e044da26cd00fc74e367c66e63/include/rootston/config.h#L41-L45) (which is part of the example compositor, so perhaps wlroots requires the compositor to handle this itself?).

Reply:
In wlc (the current compositor framework Way Cooler uses) that information is not exposed (or at least, not anywhere I can see). It, along with xwayland stuff, is all handled internally.

When we switch to wlroots though that will definitely be what we do. Also yes you're correct we have to handle it ourselves--that's the main benefit of wlroots actually is that we have to manage more stuff (compared to wlc). This increases the complexity of the compositor but also gives enough control that we can do stuff like this.